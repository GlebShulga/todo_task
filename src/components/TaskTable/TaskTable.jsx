import React, { Component } from "react";
import { fetchTaskList } from "../../redux/actions/task";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
import "./TaskTable.scss";

class TaskTable extends Component {
  constructor(props) {
    super(props);
    this.state = {
      tasks: props.tasks.taskList,
    };
  }
  componentDidMount() {
    this.props.fetchTaskList();
  }

  componentWillReceiveProps(nextProps) {
    this.setState({ tasks: nextProps.tasks.taskList });
  }

  renderTableHeader() {
    return (
      <tr>
        <th>Title</th>
        <th>Description</th>
        <th>Status</th>
      </tr>
    );
  }

  renderTableData() {
    return this.state.tasks.map((task) => {
      const { taskId, title, description, status } = task;
      return (
        <tr key={taskId}>
          <td>{title}</td>
          <td>{description}</td>
          <td>{status}</td>
        </tr>
      );
    });
  }

  render() {
    return (
      <div>
        <div className="task-table_button_position">
          <button
            className="task-table_button"
            onClick={() => {
              this.props.history.push("/");
            }}
          >
            X
          </button>
        </div>
        <h1 className="title">Tasks Table</h1>
        <table className="tasks-table">
          <tbody>
            {this.renderTableHeader()}
            {this.renderTableData()}
          </tbody>
        </table>
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  tasks: state.task,
  category: state.category
});

const mapDispatchToProps = (dispatch) => ({
    fetchTaskList: () => dispatch(fetchTaskList()),
});

export default withRouter(connect(
  mapStateToProps,
  mapDispatchToProps
)(TaskTable));
