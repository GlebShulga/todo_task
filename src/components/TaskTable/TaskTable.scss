.title {
  text-align: center;
}

.fetching {
  text-align: center;
  font-size: large;
}

.tasks-table {
  text-align: center;
  border-collapse: collapse;
  border: 0.2rem solid #ddd;
  width: 100%;
}

.tasks-table td,
.tasks-table th {
  border: 0.1rem solid #ddd;
  padding: 0.5rem;
}

.tasks-table th {
  padding: 1rem 0;
  text-align: center;
  background-color: #4caf50;
  color: white;
}

$primary: gray;

.task-table_button {
  padding: 0.4rem 0.8rem;
  font-size: 10px;
  font-weight: 900;
  color: #fff;
  background-color: $primary;
  border: 2px solid $primary;
  cursor: pointer;
  width: max-content;
  transition: 0.25s ease;

  &:hover {
    color: $primary;
    background-color: transparent;
  }
  &--transparent {
    color: $primary;
    background-color: transparent;
  }
}

.task-table_button_position {
  display: flex;
  justify-content: flex-end;
  padding-right: 0.5rem;
}
