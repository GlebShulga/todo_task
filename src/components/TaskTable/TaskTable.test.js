import React from "react";
import configureStore from "redux-mock-store";
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import renderer, {act} from "react-test-renderer";
import { shallow } from "enzyme";
import { toMatchDiffSnapshot } from "snapshot-diff";
import TaskTable from "./TaskTable";

expect.extend({toMatchDiffSnapshot});

const middlewares = [thunk];
let mockStore;
const mockStoreConf = configureStore(middlewares);

  describe("TaskTable Component", () => {
    beforeEach(() => {
      mockStore = mockStoreConf({
        task: {
          taskList: [
            {
              categoryId: "3",
              taskId: "1",
              title: "Just do it",
              description: "bla",
              status: "new",
              _createdAt: 1630474050422,
            },
            {
              categoryId: "3",
              taskId: "2",
              title: "Do it one more time",
              status: "new",
              description: "",
              _createdAt: 1630474050402,
            },
          ],
        },
      });
    });

    it("renders TaskTable without crashing", () => {
    shallow(
      <Router>
        <Provider store={mockStore}>
          <TaskTable />
        </Provider>
      </Router>
    );
    });

    it("TaskTable snapshot", () => {
      const component = renderer
        .create(
          <Router>
            <Provider store={mockStore}>
              <TaskTable />
            </Provider>
          </Router>
        )
        const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
      act(() => {
        component.root.findAllByType('button')[0].props.onClick()
      })
          const treeUpdate = component.toJSON();
          expect(tree).toMatchDiffSnapshot(treeUpdate);
    });
  });
