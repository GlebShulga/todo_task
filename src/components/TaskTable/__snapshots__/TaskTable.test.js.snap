// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskTable Component TaskTable snapshot 1`] = `
<div
  className="width"
>
  <div
    className="task-table_button_position"
  >
    <button
      className="task-table_button"
      onClick={[Function]}
    >
      X
    </button>
  </div>
  <h1
    className="title"
  >
    Tasks Table
  </h1>
  <table
    className="tasks-table"
  >
    <tbody>
      <tr>
        <th>
          Title
        </th>
        <th>
          Description
        </th>
        <th>
          Status
        </th>
      </tr>
      <tr>
        <td>
          Just do it
        </td>
        <td>
          bla
        </td>
        <td>
          new
        </td>
      </tr>
      <tr>
        <td>
          Do it one more time
        </td>
        <td>
          
        </td>
        <td>
          new
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`TaskTable Component TaskTable snapshot 2`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;
